### PR TITLE
Allow usage of variable and context in answer expression

### PR DESCRIPTION
**Pull request by maimoonak**
Fixes #[issue number]

**Description**
Currently the answer-expression option loading by fhirpath just evaluates the simple version of expression and does not make use of context, variables, and other context data `fhirPathEngine.evaluate(questionnaireResponse, expression.expression)`. This should use ExpressionEvaluator evaluateExpression method with all context data.

**Type**
Choose one: Feature

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
@maimoonak 
Can you please check on your local machine to see if the build is successful and the unit and Android tests pass as expected?

@omarismail94 
I'm not sure, but I'm unable to fetch the detailed log of the Kokoro build failure.
May be not having enough permissions?

**Comment by omarismail94** [comment]
@maimoonak there is a test that fails:

```
com.google.android.fhir.datacapture.QuestionnaireViewModelTest > resolveAnswerExpression() should return questionnaire item answer options for answer expression with fhirpath and variable FAILED
    com.google.common.truth.AssertionErrorWithFacts at QuestionnaireViewModelTest.kt:4032
```

can you fix that?

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt [hunk 1]
```diff
@@ -4282,6 +4282,65 @@ class QuestionnaireViewModelTest {
       )
   }
 
+  @Test
+  fun `should return questionnaire item answer options for answer expression with fhirpath and variable`() =
```
Comment (jingtang10) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt:4286]: This is a great test case!

Do we need to add a test case for `%context` as well?

Also, we never added support for `%questionnaire` and `%qitem` as specified here https://build.fhir.org/ig/HL7/sdc/expressions.html#fhirpath-supplements right? Can you create an issue for this?

**Review by jingtang10 — CHANGES_REQUESTED** [review]

File: datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt (hunk 1 above)
Comment (maimoonak) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt:4286]: Here it is https://github.com/google/android-fhir/issues/2163

Comment (jingtang10) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelTest.kt]: Thanks for changing the test case. Can you please make the test case for %context a separate test case? It's really a lot better to keep unit tests as simple and modular as possible.

**Review by jingtang10 — CHANGES_REQUESTED** [review]

**Review by jingtang10 — APPROVED** [review]

File: datacapture/src/main/java/com/google/android/fhir/datacapture/expressions/EnabledAnswerOptionsEvaluator.kt [hunk 2]
```diff
@@ -185,6 +199,9 @@ internal class EnabledAnswerOptionsEvaluator(
   // https://build.fhir.org/ig/HL7/sdc/expressions.html#x-fhir-query-enhancements
   private suspend fun resolveAnswerExpression(
     item: QuestionnaireItemComponent,
+    responseItem: QuestionnaireResponseItemComponent,
+    questionnaireResponse: QuestionnaireResponse,
+    questionnaireItemParentMap: Map<QuestionnaireItemComponent, QuestionnaireItemComponent>,
```
Comment (MJ1998) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/expressions/EnabledAnswerOptionsEvaluator.kt:204]: I guess these 2 were not needed in this API invocation

**Pull request merged**